### PR TITLE
Add routes based on new cloud config

### PR DESCRIPTION
**Pull request by karelhala**

### CODE REVIEW

**Comment by iphands** [comment]
@kruai  @karelhala
For styling reasons... I went through and remove all unneeded instances of "insights" and "advisor". I would like to do this unless it causes some trouble.

@karelhala also I would like to consider (maybe in another PR) have the sample application render on something other then on "/insights/sample" I wonder if we can make a new section for apps that have not been given a valid section yet. Like "/staging/**". Thus the sample app would be at "/staging/sample".

Does that make sense to do?

File: config/spandx.config.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 /*global module*/
 module.exports = {
     routes: {
-        '/apps/advisor/': { host: 'http://localhost:8002' },
-        '/insights/advisor': { host: 'http://localhost:8002' },
```
Comment (ryelo) [inline config/spandx.config.js]: Do we need to MacOSify these?

Comment (iphands) [reply config/spandx.config.js]: Yeah we do.
Maybe we should do this in the baked in config... So it happens without needing to do it

File: config/base.webpack.plugins.js [hunk 2]
```diff
@@ -84,20 +84,20 @@ const CopyFilesWebpackPlugin = new (require('copy-webpack-plugin'))([
...
 }]);
 plugins.push(HtmlReplaceWebpackPlugin);
 
 /**
- * Replaces any instance of RELEASE in js files with config.insightsDeployment value.
+ * Replaces any instance of RELEASE in js files with config.deploymentEnv value.
  */
 const Release = new webpack.DefinePlugin({
-    RELEASE: JSON.stringify(config.insightsDeployment)
+    RELEASE: JSON.stringify(config.release)
```
Comment (kruai) [inline config/base.webpack.plugins.js]: On line 92, you replaced `config.insightsDeployment` with `config.deploymentEnv`, but here you replaced it with `config.release`. The comment on line 97 also says it should be set to the `config.deploymentEnv` value. 
Is this correct?

Comment (karelhala) [reply config/base.webpack.plugins.js]: The comment is bad, should be `Replaces any instance of RELEASE in js files with config.release value.`

File: config/spandx.config.js (hunk 1 above)
Comment (karelhala) [reply config/spandx.config.js]: Yeah! Good point.

File: config/base.webpack.plugins.js [hunk 3]
```diff
@@ -93,14 +93,6 @@ const HtmlReplaceWebpackPlugin = new(require('html-replace-webpack-plugin'))([{
 }]);
 plugins.push(HtmlReplaceWebpackPlugin);
 
-/**
- * Replaces any instance of RELEASE in js files with config.deploymentEnv value.
- */
-const Release = new webpack.DefinePlugin({
-    RELEASE: JSON.stringify(config.release)
-});
-plugins.push(Release);
```
Comment (kruai) [inline config/base.webpack.plugins.js]: Do we want to delete Release entirely? After this commit, it's not defined in either `base.webpack.plugins.js` or `webpack.common.js`.

File: src/entry.js [hunk 4]
```diff
@@ -5,17 +5,13 @@ import { Provider } from 'react-redux';
 import { init } from './store';
 import App from './App';
 
-// exposes webpack variable RELEASE
```
Comment (dLobatog) [inline src/entry.js]: According to the guide, this variable should remain exposed

Comment (dLobatog) [reply src/entry.js]: It's true though that the variable is never used...

File: config/webpack.common.js [hunk 5]
```diff
@@ -32,5 +32,5 @@ module.exports = {
         static: path.resolve(__dirname, '../static'),
         publicPath
     },
-    insightsDeployment
```
[the remark this answers is not in the document]
Comment (karelhala) [reply config/webpack.common.js]: Release is nice to have, but if you don't want to use it you don't have to. If you use release variable it is just a bit more strict.

File: config/base.webpack.plugins.js (hunk 3 above)
Comment (karelhala) [reply config/base.webpack.plugins.js]: Yes, we don't need it at all. The `DefinePlugin` will not assign any variable so no need to have this plugin enabled at all.

File: src/entry.js [hunk 6]
```diff
@@ -5,17 +5,13 @@ import { Provider } from 'react-redux';
...
-/*eslint no-undef: "error"*/
-
 /**
  * Hooks up redux to app.
  *  https://redux.js.org/advanced/usage-with-react-router
  */
 ReactDOM.render(
     <Provider store={ init().getStore() }>
-        <Router basename={ `/${RELEASE}/platform/advisor` }>
+        <Router basename={ `./insights/advisor` }>
```
Comment (karelhala) [inline src/entry.js]: Wrong path, should be ` <Router basename={ `insights/advisor` }>`

**Pull request merged**